### ISSUE TITLE
scroll up an infiniteLoader that has scrollToIndex set bounces back to scrollToIndex

### ISSUE THREAD

**Issue by dcolens** [bvaughn/react-virtualized#998]
# repro steps:

   - have an infiniteloader with a list (or grid or ...) with scrollToIndex set to something like 10000 => the list shows starting at the corresponding line
   - scroll back up
   - after a few lines the list jumps back to 10000 (only a few lines for Firefox, a bit more for Chrome) => this is only triggered when fresh data is loaded.

# some investigation:
It looks very similar to https://github.com/bvaughn/react-virtualized/issues/595:

> recomputeGridSize is called by InfiniteLoader once its new rows load. This in turn sets the _recomputeScrollTopFlag flag to true. 

_recomputeScrollTopFlag is set to true in recomputeGridSize as follow:
```javascript
this._recomputeScrollTopFlag = scrollToRow >= 0 && rowIndex <= scrollToRow;
```
I'm suspecting that this code is only valid when scrolling down but not up. If this is correct, recomputeGridSize should use this.state.scrollDirectionVertical to compute _recomputeScrollTopFlag and the same issue likely applies to horizontal scrolling too.

**Comment by bvaughn** [comment]
Interested in contributing a new unit test that captures this behavior?

Better yet, a new test _and_ a fix? 😄

**Comment by dcolens** [comment]
Yes, actually I have a fix (because I hit this in a n app I develop). I also looked at the tests, I'm not too sure how to best test this ? Suggestions are welcome 😊

Did

⁣Sent from BlueMail ​

**Comment by bvaughn** [comment]
Look at the existing tests. There are many. Should be plenty of examples of testing the individual things your bug report mentions.

Comment out your fix. Write a test that fails. Uncomment your fix and make sure the test passes. 🙂

You're obviously welcome to submit just the fix, but it has a better chance of getting merged (and would really help us out) is there was a test too.

**Comment by dcolens** [comment]
I was asked to write a fix, I did it. 
I was asked to write tests, I did it.

I think I played fair. Not very motivating to contribute more.

**Comment by wuweiweiwu** [comment]
@dcolens The reason it was reverted was because the test failed after `Grid` was updated to use `gDSFP` 

Its on my list of TODOs. However, if you wan't to take some time to fix the test that is failing that'll be welcome :)

**Comment by mengdage** [comment]
@wuweiweiwu Hi, it seems like people are busy fixing this bug. I create https://github.com/bvaughn/react-virtualized/pull/1154 with fixed tests.

**Comment by wuweiweiwu** [comment]
@mengdage awesome! Ill take a look tonight!

**Comment by marsch** [comment]
the merge was reverted - still having the issue - any workarounds known here?

**Comment by wazcodez** [comment]
Facing the same issue. Any potential workarounds?

**Comment by ryanwilliamquinn** [comment]
Same issue for me.

**Comment by jayeshpp** [comment]
Same issue for me too.
InfiniteLoader + WindowScroller + Grid

**Comment by bobalazek** [comment]
Experiencing the same issue. Why was the merge reverted?

**Comment by AleshaOleg** [comment]
For me downgrading to 9.20.0 worked

**Comment by wangbigmountain** [comment]
Still have this issue InifiteLoader + List

**Comment by automatication** [comment]
@bvaughn Facing the same issue with InfinitLoader and Grid with 9.21.2, it would be amazing if this issue could be fixed.

**Comment by gigatesseract** [comment]
A workaround (that I am not proud of)

I was facing the same issue. I had to render 9k rows in total with a seek functionality for a particular row of interest. After seeking, scrolling up caused no issues, whereas when scrolling down, it automatically jumped to the already seeked row. (I am passing a state parameter to scrollToIndex prop which I am changing so that might rerender the list, but I could not explain why the issue does not come when scrolling up)  
I solved the problem by resetting the scrollToIndex once scroll is complete. 

```
this.setState({scrolled_index: index  + 20},  //this centers the row for my use case.    
          () =>{    
            this.setState({scrolled_index: -1})    
          });
```    

I am passing `this.state.scrolled_index` to the List component. 
I am currently working in a project with deadlines. Will get a plunker up and running as soon as possible. Will love to see a fix for this issue.
